src/snkit/network.py

```diff
@@ -5,6 +5,7 @@
 import numpy as np
 import pandas
 import shapely.errors
+import networkx as nx
```

Comment:
Let's make networkx an optional dependency:
```suggestion
try:
    import networkx as nx
    USE_NX = True
except ImportError:
    USE_NX = False
```
And then further down we can check (in `to_networkx`) before creating the graph, throw a clear error message if networkx is not available.

Reply:
I've adjusted the `to_networkx` function as follows: 

https://github.com/amanmajid/snkit/blob/6c506eea7dc6e2a54f5ef097d5d8b87c524f45a5/src/snkit/network.py#L613-L627

How's that?